DependenciesGui/DependencyModuleList.xaml.cs

```diff
@@ -40,6 +35,15 @@ public override int GetHashCode()
             return hashcode;
         }
 
+        public override bool Equals(object obj)
+        {
+            if (obj is ModuleCacheKey o)
+            {
+                return this.GetHashCode() == o.GetHashCode();
```

Comment:
Thank you for the fix.
But I believe this is not a good approach to compare using hash codes because there can be hash collisions. If you can make a pull request with direct comparisons on the properties, I'll merge it in to my branch.